tutorial/OpenTelemetry/deploy/ChatQnA.md

```diff
@@ -0,0 +1,92 @@
+
+# Table of contents
+
+1. [Telemetry Tracing with Jaeger on Gaudi](#telemetry-tracing-with-jaeger-on-gaudi)
+2. [Telemetry Metrics with Grafana on Gaudi](#telemetry-metrics-with-grafana-on-gaudi)
+
+
+## Telemetry Tracing with Jaeger on Gaudi
+
+After ChatQnA processes a question, two traces should appear along the timeline.  
+The trace for opea: ServiceOrchestrator.schedule runs on the CPU and includes seven spans, one of which represents the LLM host functions in general.  
+For LLM functions executed on Gaudi, stream requests are displayed under opea: llm_generate_stream.  
+This trace contains two spans: one for the first token and another for all subsequent tokens. 
+
+![chatqna_1req](../assets/jaeger_ui_opea_chatqna_1req.png)
+
+The first trace along the timeline is opea: ServiceOrchestrator.schedule, which operates on the CPU. 
+This trace provides insights into the orchestration and scheduling of services within the ChatQnA megaservice, highlighting the execution flow during the process.
+
+![chatqna_cpu_req](../assets/jaeger_ui_opea_chatqna_req_cpu.png)
+
+Clicking on the opea: ServiceOrchestrator.schedule trace will expand to reveal seven spans along the timeline.  
+The first span represents the main schedule function, which has minimal self-execution time, indicated in black.   
+The second span corresponds to the embedding microservice execution time, taking 33.72 ms as shown in the diagram.  
+Following the embedding is the retriever span, which took only 3.13 ms.  
+The last span captures the LLM functions on the CPU, with an execution time of 41.99 ms.  
+These spans provide a detailed breakdown of the execution flow and timing for each component within the service orchestration.  
+
+![chatqna_cpu_breakdown](../assets/jaeger_ui_opea_chatqna_cpu_breakdown.png)
+
+The second trace following the schedule trace is opea: llm_generate_stream, which operates on Gaudi, as depicted in the diagram.  
+This trace provides insights into the execution of LLM functions on Gaudi,  
+highlighting the processing of stream requests and the associated spans for token generation. 
+
+![chatqna_gaudi_req](../assets/jaeger_ui_opea_chatqna_req_gaudi.png)
+
+Clicking on the opea: llm_generate_stream trace will expand to reveal two spans along the timeline.  
+The first span represents the execution time for the first token, which took 15.12 ms in this run.  
+The second span captures the execution time for all subsequent tokens, taking 920 ms as shown in the diagram. 
+These spans provide a detailed view of the token generation process and the performance of LLM functions on Gaudi.
+
+![chatqna_gaudi_breakdown](../assets/jaeger_ui_opea_chatqna_req_breakdown_2.png)
+
+Overall, the traces on the CPU consist of seven spans and are represented as larger circles.   
+In contrast, the traces on Gaudi have two spans and are depicted as smaller circles.   
+The diagrams below illustrate a run with 16 user requests, resulting in a total of 32 traces.  

```

Comment:
did you by mistake interchange the semantics of large and small circles?
Gaudi faster than CPU ..

Did you want to compare a trace all running on CPU versus one on Gaudi?

Reply:
the  size of circle are related to the number of span instead of running time, so CPU traces indeed have larger circle. a ChatQnA request indeed has a trace on CPU and a trace on Gaudi. we could see Gaudi trace as smaller circle with higher latency in Y axis.  However, when the users requests are more than 16 at the same time. CPU latency will grow rapidly.  this is the PR related to this finding. https://github.com/opea-project/GenAIExamples/pull/1780